forui/lib/src/widgets/breadcrumb.dart

```diff
@@ -0,0 +1,284 @@
+  }
+}
+
+/// The [FBreadcrumbItem] data.
+class FBreadcrumbItemData extends InheritedWidget {
+  /// Returns the [FBreadcrumbItemData] of the [FBreadcrumb] in the given [context].
+  ///
+  /// ## Contract
+  /// Throws [AssertionError] if there is no ancestor [FBreadcrumb] in the given [context].
+  @useResult
+  static FBreadcrumbItemData of(BuildContext context) {
+    final data = context.dependOnInheritedWidgetOfExactType<FBreadcrumbItemData>();
+    assert(data != null, 'No FBreadcrumbData found in context');
+    return data!;
+  }
+
+  /// The breadcrumb's style.
+  final FBreadcrumbStyle style;
+
+  /// Creates a [FBreadcrumbItemData].
+  const FBreadcrumbItemData({
+    required this.style,
+    required super.child,
+    super.key,
+  });
+
+  @override
+  bool updateShouldNotify(FBreadcrumbItemData oldWidget) => style != oldWidget.style;
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(DiagnosticsProperty('style', style));
+  }
+}
+
+/// A breadcrumb item.
+mixin FBreadcrumbItem on Widget {
+  /// Creates a basic crumb.
+  ///
+  /// It is typically used to represent a single item in the breadcrumb.
+  static Widget of({
+    required VoidCallback onPress,
+    required Widget child,
+    bool? selected,
+    Key? key,
+  }) =>
+      _Crumb(onPress: onPress, selected: selected, key: key, child: child);
+
+  /// Creates a collapsed crumb.
+  ///
+  /// It is typically used to keep the breadcrumb compact when there are too many items.
+  static Widget collapsed({
+    required List<FTileGroup> menu,
+    Key? key,
+  }) =>
+      _NestedCrumbs(menu: menu, key: key);
+}
+
+class _Crumb extends StatelessWidget implements FBreadcrumbItem {
+  final bool selected;
+  final VoidCallback onPress;
+  final Widget child;
+
+  const _Crumb({
+    required this.onPress,
+    required this.child,
+    bool? selected,
+    super.key,
+  }) : selected = selected ?? false;
+
+  @override
+  Widget build(BuildContext context) {
+    final style = FBreadcrumbItemData.of(context).style;
+
+    return FTappable(
+      onPress: onPress,
+      child: DefaultTextStyle(
+        style: selected ? style.selectedTextStyle : style.unselectedTextStyle,
+        child: child,
+      ),
+    );
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(FlagProperty('selected', value: selected, ifTrue: 'selected'))
+      ..add(ObjectFlagProperty.has('onPress', onPress));
+  }
+}
+
+class _NestedCrumbs extends StatefulWidget implements FBreadcrumbItem {
+  final List<FTileGroup> menu;
+
+  const _NestedCrumbs({
+    required this.menu,
+    super.key,
+  });
+
+  @override
+  State<_NestedCrumbs> createState() => _NestedCrumbsState();
+}
+
+class _NestedCrumbsState extends State<_NestedCrumbs> with SingleTickerProviderStateMixin {
```

Comment:
ould you verify a border appear when focusing works on individual breadcrumb item? I tried to focus on individual items in the preview but it didn't seem to do anything.

Also, the mouse cursor doesn't change when hovering over the item.

I also think we should change the color when hovering over the item.